### PR TITLE
Dialog for (in)equality filter

### PR DESCRIPTION
**Pull request by HanKruiger**

### CODE REVIEW

File: web/src/main/webapp/equalityFilter.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    complement: boolean;
+}
+
+export class EqualityFilterDialog extends Dialog {
+	constructor(
+		private columnDescription: ColumnDescription,
+		private rrCallback: (filter: EqualityFilterDescription) => void,
+	) {
+		super("Equality filter");
+		this.addTextField("query", "Query for:", columnDescription.kind);
```
Comment (mbudiu-vmw) [inline web/src/main/webapp/equalityFilter.ts]: Message should be "Search"

File: web/src/main/webapp/equalityFilter.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		this.addSelectField("complement", "Check for:", ["Equality", "Inequality"]);
```
Comment (mbudiu-vmw) [inline web/src/main/webapp/equalityFilter.ts]: "Compare".

File: web/src/main/webapp/dialog.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import {IHtmlElement} from "./ui"
+import {ContentsKind, ColumnDescription} from "./table"
+import {Pair} from "./util"
+
+export class DialogField {
+	html: HTMLSelectElement | HTMLInputElement;
+	type?: ContentsKind;
+}
+
+export abstract class Dialog implements IHtmlElement {
```
Comment (mbudiu-vmw) [inline web/src/main/webapp/dialog.ts]: Can you also please modify the method chooseBuckets() in histogram.ts, the line using window.prompt, to use a dialog instead?

**Comment by HanKruiger** [comment]
All issues should be solved now!

**Pull request merged**